### PR TITLE
FIX: pathlib in nbagg

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Closes (partially) #11426 #11384 where `Path` was not defined in `backed_nbagg.py`

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant

### CODE REVIEW

**Comment by ImportanceOfBeingErnest** [comment]
This should fix #11384 as well.

**Comment by jklymak** [comment]
Trying a smoketest for nbagg.  Works locally, but not sure if we install jupyter during the test build - I can imagine that'd be quite a dependency...

**Comment by anntzer** [comment]
I think installing jupyter on one of the test instances and protecting the test with a skipif-jupyter-not-installed would be fine.

File: lib/matplotlib/tests/test_backend_nbagg.py [hunk 1]
```diff
@@ -2,8 +2,14 @@
...
-import nbformat
-
+import pytest
+try:
+    import nbformat
+    nojupyter = False
+except:
+    nojupyter = True
+
+pytestmark = pytest.mark.skipif(nojupyter, reason='Jupyter not installed')
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_backend_nbagg.py]: `nbformat = pytest.importorskip('nbformat')`

Comment (jklymak) [reply lib/matplotlib/tests/test_backend_nbagg.py]: Slick!

**Comment by jklymak** [comment]
OK, the test actually installed and worked fine on the 3.6 test, so now we have a way to smoketest that matplotlib runs on a notebook.   Open to input on where to put the notebook file, for now I just stuck it in `lib/matplotlib/tests/test_nbagg_01.ipynb`

**Comment by jklymak** [comment]
The Appveyor failure is just a timeout, which I'm 98%  certain was just a glitch...

**Pull request merged**